bin/plugin/lib/version.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+// @ts-ignore
+const semver = require( 'semver' );
+
+/**
+ * Follow the WordPress version guidelines to compute
+ * the version to be used By default, increase the "minor"
+ * number but if we reach 9, bump to the next major.
+ *
+ * @param {string} version Current version.
+ * @return {string} Next Major Version.
+ */
+function getNextMajorVersion( version ) {
```

Comment:
Seems like a nice / straight-forward thing to test.

Which, in part, goes to my second point: It might also be easy enough to implement this without the `semver` package, supported by unit tests to verify the expected behavior.

In fact, there's a (duplicate) implementation of this in `@wordpress/project-management-automation`:

https://github.com/WordPress/gutenberg/blob/bab1bc82c1382a73f81d2158532e2b1881196692/packages/project-management-automation/lib/add-milestone.js#L122

https://github.com/WordPress/gutenberg/blob/bab1bc82c1382a73f81d2158532e2b1881196692/packages/project-management-automation/lib/add-milestone.js#L140-L145